templates/nginx/nginx-config.yaml

```
@@ -31,7 +31,7 @@ data:
       access_log   /dev/stderr  main;
       sendfile     on;
       tcp_nopush   on;
-      resolver {{ default (printf "kube-dns.kube-system.svc.%s" .Values.clusterDomain ) .Values.nginx.config.dnsResolver }};
+      resolver {{ default (printf "kube-dns.kube-system.svc.%s" .Values.clusterDomain ) .Values.nginx.config.dnsResolver }} valid={{ default "10s" .Values.nginx.config.dnsTtl }};
```

The solution seems sound to me. I have not exercised this feature of nginx but I trust your testing.

I don't think we need double-defined defaults here, since you already define a default in the values.yaml.
```suggestion
      resolver {{ default (printf "kube-dns.kube-system.svc.%s" .Values.clusterDomain ) .Values.nginx.config.dnsResolver }}{{ with .Values.nginx.config.dnsTTL }} valid={{ . }}{{ end }};
```